File: tests/fixtures/ruby/sample.rb
require "json"
require_relative "helpers"

MAX_RETRIES = 3

# A base animal class
class Animal
  KINGDOM = "Animalia"

  # Create a new animal
  def initialize(name, sound)
    @name = name
    @sound = sound
  end

  # Make the animal speak
  def speak
    puts "#{@name} says #{@sound}"
  end
end

# A dog inherits from Animal
class Dog < Animal
  def speak
    puts "#{@name} barks!"
  end

  def fetch(item)
    speak
    puts "Fetching #{item}"
  end

  def self.breed_info
    lookup_breeds
  end
end

# Utility module
module Formatter
  def self.format_name(name)
    name.strip.capitalize
  end

  def self.format_list(items)
    items.map { |i| format_name(i) }.join(", ")
  end
end

def greet(name)
  formatted = Formatter.format_name(name)
  puts "Hello, #{formatted}"
end

File: tests/fixtures/python/sample.py
"""Sample Python module for testing."""

MAX_RETRIES: int = 3
DEFAULT_TIMEOUT = 30

def add(a: int, b: int) -> int:
    """Adds two numbers together."""
    return a + b

def greet(name: str) -> str:
    result = format_name(name)
    return f"Hello, {result}"

class Animal:
    """Base animal class."""
    MAX_AGE = 100

    def __init__(self, name: str, sound: str) -> None:
        self.name = name
        self.sound = sound

    def speak(self) -> str:
        """Make the animal speak."""
        return f"{self.name} says {self.sound}"

class Dog(Animal):
    """A dog that inherits from Animal."""

    def speak(self) -> str:
        return f"{self.name} barks!"

    def fetch(self, item: str) -> str:
        result = self.speak()
        print(result)
        return f"Fetching {item}"

@staticmethod
def standalone_decorated() -> None:
    pass
